spec/order_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'order'
+#
+describe Order do
+  let(:menu) { double(:menu, dishes: dishes)}
+  let(:messager) { double(:messager)}
+  let(:dishes) { {menuitem1: 1, menuitem2: 1.50 } }
+
+  subject(:order) { described_class.new(menu_instance: menu, messager: messager)}
+
+  describe '#add' do
+    it 'stores name and quantity of dish in basket' do
+      order.add(:menuitem1, 2)
+      expect(order.basket).to include(menuitem1: 2)
+    end
+  end
+
+  describe '#overview' do
+    it 'returns an empty basket when empty' do
+      expect(order.overview).to eq 'Basket is empty'
+    end
+    it 'returns content of the basket' do
+      order.add :menuitem1, 3
+      expect(order.overview).to include 'menuitem1 x3'
+    end
+  end
+
+  describe '#checkout' do
+    before do
+      msg = "menuitem1 x3: £3\nTotal price: £3"
+      allow(messager).to receive(:new).with(msg).and_return("menuitem1 x3: £3\nTotal price: £3")
```

Comment:
Metrics/LineLength: Line is too long. [96/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)